pkg/machines/components/networks/utils.js

```diff
@@ -229,3 +229,34 @@ export function isIpv6InNetwork(network, prefix, ip) {
 
     return network == ip;
 }
+
+/* If you take an IP address an apply the network mask (bitwise and) then you get the network address - no host can have such address
+ * This function returns the first usable address of the block
+ * @param {string} ipv4
+ * @param {string} netmask
+ */
+export function ipv4FirstUsableAddress(ipv4, netmask) {
+    // Within a subnet, two host addresses - all-zeros and one all-ones are reserved as network address and broadcast, respectively.
+    if (!ipv4)
+        return undefined;
+
+    let mask = netmask;
+    if (netmask.split(".").length == 4)
+        mask = netmaskToCIDR(netmask);
```

Comment:
Can it happen that ipv4 is in an invalid format here? like, less or more than 4 components, numbers <1 or > 254, or not numbers at all, etc? I. e. does this function need to do that structural validation, or is that already done elsewhere?